src/formatter/helpers/formatters.ts

```diff
@@ -0,0 +1,52 @@
+import Formatter from '../.'
+import JsonFormatter from '../json_formatter'
+import MessageFormatter from '../message_formatter'
+import ProgressBarFormatter from '../progress_bar_formatter'
+import ProgressFormatter from '../progress_formatter'
+import RerunFormatter from '../rerun_formatter'
+import SnippetsFormatter from '../snippets_formatter'
+import SummaryFormatter from '../summary_formatter'
+import UsageFormatter from '../usage_formatter'
+import UsageJsonFormatter from '../usage_json_formatter'
+import HtmlFormatter from '../html_formatter'
+
+const Formatters = {
+  getFormatters(): Record<string, typeof Formatter> {
+    return {
+      json: JsonFormatter,
+      message: MessageFormatter,
+      html: HtmlFormatter,
+      progress: ProgressFormatter,
+      'progress-bar': ProgressBarFormatter,
+      rerun: RerunFormatter,
+      snippets: SnippetsFormatter,
+      summary: SummaryFormatter,
+      usage: UsageFormatter,
+      'usage-json': UsageJsonFormatter,
+    }
+  },
+  buildFormattersDocumentationString(): string {
+    let concatanatedFormattersDocumentation: string = ''
+    const formattersDocumentation: Record<string, string> = {
+      json: JsonFormatter.documentation,
+      message: MessageFormatter.documentation,
+      html: HtmlFormatter.documentation,
+      progress: ProgressFormatter.documentation,
+      'progress-bar': ProgressBarFormatter.documentation,
+      rerun: RerunFormatter.documentation,
+      snippets: SnippetsFormatter.documentation,
+      summary: SummaryFormatter.documentation,
+      usage: UsageFormatter.documentation,
+      'usage-json': UsageJsonFormatter.documentation,
+    }
+
+    for (const formatter in formattersDocumentation) {
+      concatanatedFormattersDocumentation +=
+        formatter + ' : ' + formattersDocumentation[formatter] + '\n'
+    }
+
+    return concatanatedFormattersDocumentation
```

Comment:
You could use getFormatters and do something like this:

```typescript
const formatters = getFormatters()

for (let formatterKey in formatters) {
  concatanetedFormattersDocumentation += formatterKey + ': ' + formatters[formatterKey].documentation
}
```

(I've not tested this, this is a general though ;))

Reply:
Wouldn't this be a problem since the **getFormatters** method return typeof Formatter and not the formatter itself? It seems that some sort of casting is required. 

I.E.

`if (formatters[formatterName] instanceof JsonFormatter) `

I tried adding this code and got the error that:

`Property 'documentation' does not exist on type 'typeof Formatter'.ts(2339)`

Reply:
We may have to add the documentation property to the formatter type definition

Reply:
I think I already did that here (in a previous commit):

```
export default class Formatter {
  protected colorFns: IColorFns
  protected cwd: string
  protected eventDataCollector: EventDataCollector
  protected log: IFormatterLogFn
  protected snippetBuilder: StepDefinitionSnippetBuilder
  protected stream: WritableStream
  protected supportCodeLibrary: ISupportCodeLibrary
  private readonly cleanup: IFormatterCleanupFn
  protected readonly documentation: string
```

Reply:
I think the `static` attribute is missing